api/src/main/java/com/epam/pipeline/manager/datastorage/providers/nfs/NFSQuotasMonitor.java

```diff
@@ -0,0 +1,209 @@
+        this.emailNotificationAction = emailNotificationAction;
+        this.readOnlyAction = readOnlyAction;
+        this.disableMountAction = disableMountAction;
+        this.notificationsKey = notificationsKey;
+    }
+
+    @Scheduled(fixedDelayString = "${data.storage.nfs.quota.poll:60000}")
+    public void controlQuotas() {
+        log.info("Start NFS quotas processing...");
+        final List<NFSDataStorage> nfsDataStorages = loadAllNFS();
+        final Map<Long, NFSQuota> activeQuotas = loadStorageQuotas(nfsDataStorages);
+        nfsDataStorages.forEach(storage -> {
+            final NFSStorageMountStatus statusUpdate = Optional.ofNullable(activeQuotas.get(storage.getId()))
+                .map(quota -> processActiveQuota(quota, storage))
+                .orElse(NFSStorageMountStatus.ACTIVE);
+            dataStorageManager.updateMountStatus(storage, statusUpdate);
+        });
+        log.info("NFS quotas are processed successfully.");
+    }
+
+    private List<NFSDataStorage> loadAllNFS() {
+        return dataStorageManager.getDataStorages().stream()
+            .filter(storage -> DataStorageType.NFS.equals(storage.getType()))
+            .map(NFSDataStorage.class::cast)
+            .collect(Collectors.toList());
+    }
+
+    private NFSStorageMountStatus processActiveQuota(final NFSQuota quota, final NFSDataStorage storage) {
+        return CollectionUtils.emptyIfNull(quota.getNotifications()).stream()
+            .filter(Objects::nonNull)
+            .sorted(Comparator.comparing(NFSQuotaNotificationEntry::getValue).reversed())
+            .filter(this::hasRestrictingActions)
+            .filter(notification -> excessLimit(storage, notification))
+            .map(notification -> mapNotificationToStatus(storage.getId(), notification))
+            .findFirst()
+            .orElse(NFSStorageMountStatus.ACTIVE);
+    }
+
+    private NFSStorageMountStatus mapNotificationToStatus(final Long storageId,
+                                                          final NFSQuotaNotificationEntry notification) {
+        final Set<String> actions = notification.getActions();
+        if (actions.contains(disableMountAction)) {
+            return NFSStorageMountStatus.MOUNT_DISABLED;
+        } else if (actions.contains(readOnlyAction)) {
+            return NFSStorageMountStatus.READ_ONLY;
+        } else {
+            log.warn(messageHelper.getMessage(MessageConstants.STORAGE_QUOTA_UNKNOWN_RESTRICTION, actions, storageId));
+            return NFSStorageMountStatus.READ_ONLY;
+        }
+    }
+
+    private boolean hasRestrictingActions(final NFSQuotaNotificationEntry notification) {
+        final Set<String> actions = notification.getActions();
+        return CollectionUtils.size(actions) > 1
+               || (CollectionUtils.size(actions) == 1 && !actions.contains(emailNotificationAction));
+    }
+
+    private boolean excessLimit(final NFSDataStorage storage, final NFSQuotaNotificationEntry notification) {
```

Comment:
Better name is `exceedsLimit`, same for `exceedsAbsoluteLimit ` and similar

Reply:
Done